lib/simctl.js

```diff
@@ -345,70 +341,77 @@ async function shutdown (udid) {
     // in some versions of Xcode it will be a patch version
+    let runtimeId;
     try {
-      runtimeId = await getRuntimeForPlatformVersion(runtimeId, platform);
+      runtimeId = await getRuntimeForPlatformVersion(platformVersion, platform);
     } catch (err) {
-      log.warn(`Unable to find runtime for iOS '${runtimeId}'. Continuing`);
+      log.warn(`Unable to find runtime for iOS '${platformVersion}'. Continuing`);
+      runtimeId = platformVersion;
     }
 
-    const clangVersion = await getClangVersion();
-    // 1st comparison: clangVersion
-    // Command line tools version is 10.2+, but xcode 10.1 can happen
-    let isNewerIdFormatRequired = clangVersion && util.compareVersions(clangVersion, '>=',
-      CMDLINE_TOOLS_CLANG_FORMAT_CHANGED_VERSION);
-
-    if (!isNewerIdFormatRequired) {
-      // 2nd comparison: getVersion
-      // The opposite can also happen,
-      // but the combination of 10.2 command line tools and lower Xcode version happens more frequently
-      const xcodeVersion = await getVersion(false);
-      isNewerIdFormatRequired = xcodeVersion && util.compareVersions(xcodeVersion, '>=',
-        XCODE_FORMAT_CHANGED_VERSION);
+    // get the possible runtimes, which will be iterated over
+    // compute the `major.minor` version first since Xcode usually has a runtime
+    // that does not include the patch version, even if it exists
+    const runtimeIdSemver = semver.coerce(runtimeId);
+    if (_.isNil(runtimeIdSemver)) {
+      throw new Error(`Unable to parse runtime id '${runtimeId}'`);
     }
 
-    if (isNewerIdFormatRequired) {
-      runtimeId = `${SIM_RUNTIME_NAME}${platform}-${runtimeId.replace(/\./g, '-')}`;
+    let potentialRuntimeIds = [`${runtimeIdSemver.major}.${runtimeIdSemver.minor}`];
+    if (runtimeId.split('.').length === 3) {
+      potentialRuntimeIds.push(runtimeId);
+    }
+    for (const id of potentialRuntimeIds) {
+      // push the modified version first, since on modern Xcodes this is the one
+      // that is used
+      runtimeIds.push(`${SIM_RUNTIME_NAME}${platform}-${id.replace(/\./g, '-')}`);
     }
+    runtimeIds.push(...potentialRuntimeIds);
   }
 
-  log.debug(`Creating simulator with name '${name}', device type id '${deviceTypeId}' and runtime id '${runtimeId}'`);
   let udid;
-  try {
-    const out = await simExec('create', 0, [name, deviceTypeId, runtimeId]);
-    udid = out.stdout.trim();
-  } catch (err) {
-    let reason = err.message;
-    if (err.stderr) {
-      reason = err.stderr.trim();
+  for (const runtimeId of runtimeIds) {
+    log.debug(`Creating simulator with name '${name}', device type id '${deviceTypeId}' and runtime id '${runtimeId}'`);
+    try {
+      const out = await simExec('create', 0, [name, deviceTypeId, runtimeId]);
+      udid = out.stdout.trim();
+      break;
+    } catch (err) {
+      const reason = err.stderr ? err.stderr.trim() : err.message;
+      log.warn(`Unable to create simulator: ${reason}`);
```

Comment:
Do we need to print these warnings? we anyway know some of these ids won't work

Reply:
Perhaps it only makes sense to print them if the whole sim creation op fails

Reply:
I think it would be confusing if it were not there, as there would be multiple lines saying it was creating a simulator, with no indication of why.

Reply:
Played around a bit and the error gets logged elsewhere, so the confusion I envisioned is not there.